test/filter.test.ts

```diff
@@ -112,4 +118,29 @@ describe("matchFilter", () => {
             );
         }
     });
+
+    it("should work with correct type-inference", () => {
+        const text = "I <3 grammY";
+        const ctx = new Context(
+            // deno-lint-ignore no-explicit-any
+            { message: { text } } as any,
+            // deno-lint-ignore no-explicit-any
+            undefined as any,
+            // deno-lint-ignore no-explicit-any
+            undefined as any,
+        );
+        const pred = matchFilter([":text", "callback_query:data"]);
+        if (pred(ctx)) {
+            if (ctx.callbackQuery) {
+                const s: string = ctx.update.callback_query.data;
+                assert(s);
+                throw "never";
```

Comment:
Perhaps you could replace this and other instances of it with https://deno.land/std@0.224.0/assert/unreachable.ts.